source/beamline/CLS/CLSExclusiveStatesControl.cpp

```diff
@@ -0,0 +1,103 @@
+#include "CLSExclusiveStatesControl.h"
+#include "beamline/AMPVControl.h"
+
+CLSExclusiveStatesControl::CLSExclusiveStatesControl(const QString &name, const QString &statusPV, const QString &openPV, const QString &closePV, QObject *parent) :
+	AMExclusiveStatesEnumeratedControl(name, "", parent)
+{
+	setStatusControl(new AMReadOnlyPVControl(QString("%1%2").arg(name).arg("Status"), statusPV, this));
+	addClosedState(new AMSinglePVControl(QString("%1%2").arg(name).arg("Close"), closePV, this), Status::Closed, 1);
+	addOpenState(new AMSinglePVControl(QString("%1%2").arg(name).arg("Open"), openPV, this), Status::Open, 1);
+	addBetweenState(Status::Between);
+}
+
+CLSExclusiveStatesControl::~CLSExclusiveStatesControl()
+{
+
+}
+
+int CLSExclusiveStatesControl::alarmStatus() const
+{
+	return control_->alarmStatus();
+}
+
+int CLSExclusiveStatesControl::alarmSeverity() const
+{
+	return control_->alarmSeverity();
+}
+
+bool CLSExclusiveStatesControl::isOpen() const
+{
+	return (int(value()) == Open);
+}
+
+bool CLSExclusiveStatesControl::isClosed() const
+{
+	return (int(value()) == Closed);
+}
+
+bool CLSExclusiveStatesControl::isBetween() const
+{
+	return (int(value()) == Between);
+}
+
+bool CLSExclusiveStatesControl::setBaseControl(AMControl *newControl)
```

Comment:
This function is alike the implementation of super class `AMSingleEnumeratedControl`. The difference is this implementation has signal connections. 

My question is whether we need the signal connections for `AMSingleEnumeratedControl` as well? If yes, this implementation can be deleted.